src/layers/XRLayer/XRLayer.js

```diff
@@ -169,12 +171,30 @@ export default class XRLayer extends Layer {
   draw({ uniforms }) {
     const { textures, model } = this.state;
     if (textures && model) {
-      const { sliderValues, colorValues, opacity } = this.props;
+      const {
+        sliderValues,
+        colorValues,
+        opacity,
+        domain,
+        dtype,
+        channelIsOn
+      } = this.props;
+      // Check number of textures not null.
+      const numTextures = Object.values(textures).filter(t => t).length;
+      // Slider values and color values can come in before textures since their data is async.
+      // Thus we pad based on the number of textures bound.
+      const { paddedSliderValues, paddedColorValues } = padColorsAndSliders({
+        sliderValues: sliderValues.slice(0, numTextures),
+        colorValues: colorValues.slice(0, numTextures),
+        channelIsOn: channelIsOn.slice(0, numTextures),
+        domain,
+        dtype
+      });
       model
         .setUniforms({
           ...uniforms,
-          colorValues,
-          sliderValues,
+          colorValues: paddedColorValues,
+          sliderValues: paddedSliderValues,
           opacity,
           ...textures
```

Comment:
General question. Is setUniforms something we have defined? or is `model` from deck.gl?

Reply:
From luma.gl